src/Orleans.Runtime/Silo/Silo.cs

```diff
@@ -644,7 +660,8 @@ private void ConfigureThreadPoolAndServicePointSettings()
         /// </summary>
         public void Stop()
         {
-            Terminate(false);
+            this.gracefully = false;
+            StopAsync(CancellationToken.None).GetAwaiter().GetResult();
```

Comment:
Shouldn't you pass a canceled token?
BTW, the `gracefully` field is immediately overwritten after calling `StopAsync`

Reply:
good catch.  I change the behavior and forgot about the other calls.